labs/9.md

```diff
@@ -9,30 +9,30 @@ We’re going to set up a flask app. And we are going to call it lab9
 
 There are 3 main component to this lab. We will install Nginx, uWSGI, and Flask through the `apt` command. Then we will git clone an example app and config files from class repository. Next, we will modify the configuration files. If configured correctly, we'll go through the startup procedure. Then end it with some debugging tips and tricks. 
 
-## Note
-
-* No sudo was included in any command examples. Use your best judgment to include sudo or not. 
-
 ### The Setup
 
 Virtualenv is a python tool that lets you create a custom python execution environment that is specific to your project. This practice is to prevent custom libraries or dependencies from polluting your main coding environment. We need virtualenv for staging. Run
 
-    apt install virtualenv
+    sudo apt install build-essential python-dev 
+    sudo apt install virtualenv
 
 Next, we'll create the directory where our project will be located. We'll just call it [lab9] and we'll place it in /opt/. 
 
-    mkdir /opt/lab9
+    sudo mkdir /opt/lab9
+    sudo chown $USER:www-data /opt/lab9/
 
 Let's create the virtualenv for this project and activate it. But first we chown the group to www-data before we setup virtualenv. That way all the new files created will inherit this ownership. The reason we give group ownership to www-data is because our Nginx and uWSGI belongs to this group.  
 
-    chown -R :www-data lab9
+    sudo chown -R :www-data lab9
     cd /opt/lab9
     virtualenv venv
     source venv/bin/activate
     
 
 You can see a (venv) in the beginning of your prompt. Now all your python code will execute the venv version of python. Located in "/opt/lab9/venv/bin/python". It will add venv to your python path, that means python execuation procedure will search venv first before search global library directory. 
 
+To exit the virtual environment, type 
```

Comment:
Did you forget to add something here like `deactivate`?

Reply:
Good eye! corrected